### PR TITLE
Added unit tests for service_manager class

### PR DESCRIPTION
**Pull request by piyushghai**
Unit tests for service_manager.py
Unit test cases for MXNET model service error
Unit test cases for model_service.py

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: mms/tests/unit_tests/test_service_manager.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    result = service_manager.parse_modelservices_from_module(service_file_path)
+
+    class_list = [x.__name__ for x in result]
+
+    assert "SomeOtherClass" not in class_list
+    assert "SingleNodeService" in class_list
+    assert "DummyNodeService" in class_list
+
+
+@patch.object(service_manager, 'get_modelservices_registry', wraps=service_manager.get_modelservices_registry)
```
[the remark this answers is not in the document]
Comment (piyushghai) [reply mms/tests/unit_tests/test_service_manager.py]: I tend to prefer the decorators, for they make for easier reading of the test function and also enlist all the dependencies for that on the signature of the function itself. 
Passing in a mocker as a parameter can be a little hard to comprehend if one reads the code for the first time. :)

File: mms/tests/unit_tests/test_service_manager.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+
+@patch.object(service_manager,'get_registered_modelservices',
+              wraps=service_manager.get_registered_modelservices)
+@patch.object(service_manager, 'load_model', wraps=service_manager.load_model)
+def test_register_and_load_modules(load_model, registered_modelservices):
+    model_name = model_names[0]
+    model_dir = 'mms/tests/unit_tests/test_utils'
+    manifest = {}
+    module_file_path = ''.join([model_dir, '/dummy_model_service.py'])
```
Comment (vrakesh) [inline mms/tests/unit_tests/test_service_manager.py]: os.path.join, since it is a path

File: mms/tests/unit_tests/test_service_manager.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+    assert "Error when loading service file" in error.value.args[0]
+
+
+def test_parse_modelservices_from_module_with_correct_file():
+    service_file_path = 'mms/tests/unit_tests/test_utils/dummy_model_service.py'
+
+    result = service_manager.parse_modelservices_from_module(service_file_path)
+
+    class_list = [x.__name__ for x in result]
```
Comment (vrakesh) [inline mms/tests/unit_tests/test_service_manager.py]: use a set maybe

File: mms/tests/unit_tests/test_mxnet_model_service_error.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+from mms.utils.model_server_error_codes import ModelServerErrorCodes
+
+
+def test_get_code():
+    error = MMSError(ModelServerErrorCodes.INVALID_LOAD_MESSAGE, "invalid load")
+    assert error.get_code() == ModelServerErrorCodes.INVALID_LOAD_MESSAGE
+
+
+def test_get_message():
+    error = MMSError(ModelServerErrorCodes.INVALID_LOAD_MESSAGE, "invalid load")
```
[the remark this answers is not in the document]
Comment (piyushghai) [reply mms/tests/unit_tests/test_mxnet_model_service_error.py]: Seems fair. Will do it.

File: mms/tests/unit_tests/test_service_manager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (piyushghai) [reply mms/tests/unit_tests/test_service_manager.py]: Discussed offline. We should be using mockers with fixtures if they are to be reused across functions and decorators don't work across functions...

File: mms/tests/unit_tests/test_service_manager.py (hunk 3 above)
Comment (piyushghai) [reply mms/tests/unit_tests/test_service_manager.py]: Done.

File: mms/tests/unit_tests/test_service_manager.py (hunk 2 above)
Comment (piyushghai) [reply mms/tests/unit_tests/test_service_manager.py]: Done.

File: mms/tests/unit_tests/test_service_manager.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    service_file_path = 'mms/tests/unit_tests/test_utils/dummy_model_service.py'
+    service_manager.register_module(service_file_path)
+
+    add_modelservice.assert_called()
+    parse_modelservice.assert_called_with(service_file_path)
+
+
+@patch.object(service_manager, 'register_module', wraps=service_manager.register_module)
+def test_register_and_load_modules_with_assertion_error(spy):
+    model_name = model_names[0]
```
[the remark this answers is not in the document]
Comment (piyushghai) [reply mms/tests/unit_tests/test_service_manager.py]: Done.

File: mms/tests/unit_tests/test_service_manager.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+    response = service_manager.unload_models(model_names[0])
+    del (expected_response[model_names[0]])
+
+    assert model_names[0] not in response
+    assert expected_response == response
+
+
+@patch.object(service_manager, 'parse_modelservices_from_module', wraps=service_manager.parse_modelservices_from_module)
+def test_register_module_with_errors(parse_modelservices):
```
[the remark this answers is not in the document]
Comment (piyushghai) [reply mms/tests/unit_tests/test_service_manager.py]: Done. Good suggestion.

File: mms/tests/unit_tests/test_service_manager.py [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    assert response['resnet'] == 'resnet.py'
+
+
+def test_get_modelservices_registry_with_null_model_names():
+    response = service_manager.get_modelservices_registry()
+
+    assert response.name == 'modelservice'
+
+
+def test_modelservices_registry_with_model_names():
```
[the remark this answers is not in the document]
Comment (piyushghai) [reply mms/tests/unit_tests/test_service_manager.py]: Done.

File: mms/tests/unit_tests/test_mxnet_model_service_error.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (piyushghai) [reply mms/tests/unit_tests/test_mxnet_model_service_error.py]: Done.

File: mms/tests/unit_tests/test_mxnet_model_service_error.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+def test_set_code(mms_error):
+    mms_error.set_code(ModelServerErrorCodes.INVALID_PREDICT_MESSAGE)
+
+    assert mms_error.get_code() == ModelServerErrorCodes.INVALID_PREDICT_MESSAGE
+
+
+def test_set_message(mms_error):
+    mms_error.set_message("invalid_load_message")
+
+    assert mms_error.get_message() == "invalid_load_message"
```
Comment (frankfliu) [inline mms/tests/unit_tests/test_mxnet_model_service_error.py]: No new line at the end of the file.

File: mms/tests/unit_tests/test_utils/dummy_model_service.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    def signature(self):
+        pass
+
+    def ping(self):
+        pass
+
+
+class SomeOtherClass:
+    def __init__(self):
+        pass
```
Comment (frankfliu) [inline mms/tests/unit_tests/test_utils/dummy_model_service.py]: no eol

File: ci/Dockerfile.python2.7 [hunk 10]
```diff
@@ -179,6 +179,7 @@ RUN set -ex \
     && pip install retrying \
     && pip install mock \
     && pip install pytest -U \
+    && pip install pytest-mock -U \
```
[the remark this answers is not in the document]
Comment (piyushghai) [reply ci/Dockerfile.python2.7]: Will remove it.

File: mms/tests/unit_tests/test_utils/dummy_model_service.py (hunk 9 above)
Comment (piyushghai) [reply mms/tests/unit_tests/test_utils/dummy_model_service.py]: Done.

File: mms/tests/unit_tests/test_mxnet_model_service_error.py (hunk 8 above)
Comment (piyushghai) [reply mms/tests/unit_tests/test_mxnet_model_service_error.py]: Done.

**Pull request merged**